# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: f441cb5e7e6e
# Full Hash: f441cb5e7e6e30d4a60b08ca99e56a29d3dc3a8b
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-01 09:40:15
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r cfae5efaae37 -r f441cb5e7e6e dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Mon Mar 31 19:49:02 2025 +0000
+++ b/dom/workers/WorkerScope.h	Mon Mar 31 19:49:02 2025 +0000
@@ -86,6 +86,7 @@
 class VsyncWorkerChild;
 class WebTaskScheduler;
 class WebTaskSchedulerWorker;
+class WebTaskSchedulingState;
 struct RequestInit;
 
 namespace cache {
@@ -385,6 +386,11 @@
 
   WebTaskScheduler* Scheduler();
   WebTaskScheduler* GetExistingScheduler() const;
+  void SetWebTaskSchedulingState(WebTaskSchedulingState* aState) override;
+
+  WebTaskSchedulingState* GetWebTaskSchedulingState() const override {
+    return mWebTaskSchedulingState;
+  }
 
   bool WindowInteractionAllowed() const;
 
@@ -421,6 +427,7 @@
   RefPtr<cache::CacheStorage> mCacheStorage;
   RefPtr<DebuggerNotificationManager> mDebuggerNotificationManager;
   RefPtr<WebTaskSchedulerWorker> mWebTaskScheduler;
+  RefPtr<WebTaskSchedulingState> mWebTaskSchedulingState;
   RefPtr<TrustedTypePolicyFactory> mTrustedTypePolicyFactory;
   uint32_t mWindowInteractionsAllowed = 0;
   bool mIsEligibleForMessaging{true};